pallets/chainbridge/src/tests.rs

```diff
@@ -0,0 +1,537 @@
+
+		// Attempt to execute
+		assert_ok!(Bridge::eval_vote_state(
+			Origin::signed(RELAYER_A),
+			prop_id,
+			src_id,
+			Box::new(proposal.clone())
+		));
+
+		let prop = Bridge::votes(src_id, (prop_id.clone(), proposal.clone())).unwrap();
+		let expected = ProposalVotes {
+			votes_for: vec![RELAYER_A],
+			votes_against: vec![],
+			status: ProposalStatus::Approved,
+			expiry: ProposalLifetime::get() + 1,
+		};
+		assert_eq!(prop, expected);
+
+		assert_eq!(Balances::free_balance(RELAYER_B), 0);
+		assert_eq!(
+			Balances::free_balance(Bridge::account_id()),
+			ENDOWED_BALANCE
+		);
+
+		assert_events(vec![
+			Event::Bridge(PalletEvent::VoteFor(src_id, prop_id, RELAYER_A)),
+			Event::Bridge(PalletEvent::RelayerThresholdChanged(1)),
+			Event::Bridge(PalletEvent::ProposalApproved(src_id, prop_id)),
+			Event::Bridge(PalletEvent::ProposalSucceeded(src_id, prop_id)),
+		]);
+	})
+}
+
+#[test]
+fn proposal_expires() {
+	let src_id = 1;
+	let r_id = derive_resource_id(src_id, b"remark");
+
+	new_test_ext_initialized(src_id, r_id, b"System.remark".to_vec()).execute_with(|| {
+		let prop_id = 1;
+		let proposal = make_proposal(vec![10]);
+
+		// Create proposal (& vote)
+		assert_ok!(Bridge::acknowledge_proposal(
+			Origin::signed(RELAYER_A),
+			prop_id,
+			src_id,
+			r_id,
+			Box::new(proposal.clone())
+		));
+		let prop = Bridge::votes(src_id, (prop_id.clone(), proposal.clone())).unwrap();
+		let expected = ProposalVotes {
+			votes_for: vec![RELAYER_A],
+			votes_against: vec![],
+			status: ProposalStatus::Initiated,
+			expiry: ProposalLifetime::get() + 1,
+		};
+		assert_eq!(prop, expected);
+
+		// Increment enough blocks such that now == expiry
+		System::set_block_number(ProposalLifetime::get() + 1);
+
+		// Attempt to submit a vote should fail
+		assert_noop!(
+			Bridge::reject_proposal(
+				Origin::signed(RELAYER_B),
+				prop_id,
+				src_id,
+				r_id,
+				Box::new(proposal.clone())
+			),
+			Error::<Test>::ProposalExpired
+		);
+
+		// Proposal state should remain unchanged
+		let prop = Bridge::votes(src_id, (prop_id.clone(), proposal.clone())).unwrap();
+		let expected = ProposalVotes {
+			votes_for: vec![RELAYER_A],
+			votes_against: vec![],
+			status: ProposalStatus::Initiated,
+			expiry: ProposalLifetime::get() + 1,
+		};
+		assert_eq!(prop, expected);
+
+		// eval_vote_state should have no effect
+		assert_noop!(
+			Bridge::eval_vote_state(
+				Origin::signed(RELAYER_C),
+				prop_id,
+				src_id,
+				Box::new(proposal.clone())
+			),
+			Error::<Test>::ProposalExpired
+		);
+		let prop = Bridge::votes(src_id, (prop_id.clone(), proposal.clone())).unwrap();
+		let expected = ProposalVotes {
+			votes_for: vec![RELAYER_A],
+			votes_against: vec![],
+			status: ProposalStatus::Initiated,
+			expiry: ProposalLifetime::get() + 1,
+		};
+		assert_eq!(prop, expected);
+
+		assert_events(vec![Event::Bridge(PalletEvent::VoteFor(
+			src_id, prop_id, RELAYER_A,
+		))]);
+	})
+}
```

Comment:
Please add empty line at the end of file.

Reply:
done